setup autolinking for packages in bare-expo (#11403)
expo/expo: packages/expo-yarn-workspaces/README.md

The `expo-yarn-workspaces` package defines a Metro configuration object that makes Metro work with Yarn workspaces in the Expo repo. It configures Metro to include packages from the workspace root, resolves symlinked packages, excludes modules from Haste's module system, and exclude modules in the native Android and Xcode projects. You can further customize this configuration object before exporting it, if needed.

**Aside:** when starting the project, run `expo start --clear` so Metro uses the latest configuration instead of working with cached values.

### Configuration

> You can configure workspaces using the `workspaces` object in the `package.json`.

Reviewer: ```suggestion
You can configure workspaces using the `workspace` field in each workspace package's package.json file.
```

> You can configure workspaces using the `workspace` field in each workspace package's package.json file.

Reviewer: ```suggestion
You can configure workspaces using the `expo-yarn-workspaces` field in each workspace package's package.json file.
```

#### symlinks

> Sometimes a node module must be located in the project's `node_modules` folder for things to work properly. Using the `workspaces.symlinks` string array you can define a list of packages that should be linked on install.

Reviewer: ```suggestion
Sometimes an npm package must be located in the project's `node_modules` folder for things to work properly. Using the `workspaces.symlinks` string array you can define a list of packages to symlink under the project's `node_modules` folder after installing the workspaces' dependencies.

> Sometimes an npm package must be located in the project's `node_modules` folder for things to work properly. Using the `workspace.symlinks` string array you can define a list of packages to symlink under the project's `node_modules` folder after installing the workspaces' dependencies.

Reviewer: ```suggestion
Sometimes an npm package must be located in the project's `node_modules` folder for things to work properly. Using the `expo-yarn-workspace.symlinks` string array you can define a list of packages to symlink under the project's `node_modules` folder after installing the workspaces' dependencies.
```
